examples/src/main/java/mil/nga/giat/geowave/examples/query/SpatialQueryExample.java

```diff
@@ -0,0 +1,497 @@
+package mil.nga.giat.geowave.examples.query;
+
+import com.vividsolutions.jts.geom.Coordinate;
+import com.vividsolutions.jts.geom.Geometry;
+import com.vividsolutions.jts.geom.Point;
+import com.vividsolutions.jts.io.ParseException;
+import com.vividsolutions.jts.io.WKTReader;
+import mil.nga.giat.geowave.adapter.vector.FeatureDataAdapter;
+import mil.nga.giat.geowave.adapter.vector.VectorDataStore;
+import mil.nga.giat.geowave.core.geotime.GeometryUtils;
+import mil.nga.giat.geowave.core.geotime.IndexType;
+import mil.nga.giat.geowave.core.geotime.store.query.SpatialQuery;
+import mil.nga.giat.geowave.core.index.ByteArrayId;
+import mil.nga.giat.geowave.core.store.CloseableIterator;
+import mil.nga.giat.geowave.datastore.accumulo.BasicAccumuloOperations;
+import mil.nga.giat.geowave.datastore.accumulo.metadata.AccumuloAdapterStore;
+import org.apache.accumulo.core.client.AccumuloException;
+import org.apache.accumulo.core.client.AccumuloSecurityException;
+import org.apache.accumulo.core.client.Connector;
+import org.apache.accumulo.core.client.mock.MockInstance;
+import org.apache.accumulo.core.client.security.tokens.PasswordToken;
+import org.apache.log4j.Logger;
+import org.geotools.feature.AttributeTypeBuilder;
+import org.geotools.feature.simple.SimpleFeatureBuilder;
+import org.geotools.feature.simple.SimpleFeatureTypeBuilder;
+import org.geotools.filter.text.cql2.CQL;
+import org.geotools.filter.text.cql2.CQLException;
+import org.geotools.geometry.jts.JTSFactoryFinder;
+import org.opengis.feature.simple.SimpleFeature;
+import org.opengis.feature.simple.SimpleFeatureType;
+
+import java.util.ArrayList;
+
+/**
+ * This class is intended to provide a few examples on running Geowave queries of different types:
+ * 1- Querying by polygon a set of points.
+ * 2- Filtering on attributes of features using CQL queries
+ * 3- Ingesting polygons, and running polygon intersect queries.
+ * You can check all points, geometries and query accuracy in a more visual manner @ http://geojson.io/
+ */
+public class SpatialQueryExample {
+    private static Logger log = Logger.getLogger(SpatialQueryExample.class);
+
+	//We'll use GeoWave's VectorDataStore, which allows to run CQL rich queries
+	private static VectorDataStore dataStore;
```

Comment:
Although you can use this interface now.  It is short lived.  It will  not be part of the API in 0.9